src/pages/main.test.tsx

```diff
@@ -13,6 +46,113 @@ describe('Main page', () => {
-    expect(screen.getByTestId('items-cards-list')).toBeInTheDocument();
+    expect(screen.getByAltText('Loading...')).toBeInTheDocument();
+    expect(await screen.findByTestId('items-cards-list')).toBeInTheDocument();
+    expect(screen.queryByAltText('Loading...')).not.toBeInTheDocument();
+    expect(screen.getAllByTestId('item-card').length).toBe(itemsData.length);
+    expect(screen.queryByTestId('err-msg')).not.toBeInTheDocument();
+    expect(fakeFetch).lastCalledWith(
+      'https://the-one-api.dev/v2/character/?page=1&limit=10&name=//i',
+      options
+    );
+  });
+
+  it('submits search with results by Enter key pressing in SearchBar', async () => {
+    render(
+      <BrowserRouter>
+        <Main />
+      </BrowserRouter>
+    );
+    expect(await screen.findByTestId('items-cards-list')).toBeInTheDocument();
+    userEvent.type(screen.getByRole('textbox'), '{selectall}Search with result{Enter}');
+    expect(await screen.findByAltText('Loading...')).toBeInTheDocument();
+    expect(await screen.findByTestId('items-cards-list')).toBeInTheDocument();
+    expect(screen.queryByAltText('Loading...')).not.toBeInTheDocument();
+    expect(screen.getAllByTestId('item-card').length).toBe(itemsData.length);
+    expect(screen.queryByTestId('err-msg')).not.toBeInTheDocument();
+    expect(fakeFetch).toBeCalledWith(
+      'https://the-one-api.dev/v2/character/?page=1&limit=10&name=/Search with result/i',
+      options
+    );
+  });
+
+  it('submits search without results by Search button click', async () => {
+    render(
+      <BrowserRouter>
+        <Main />
+      </BrowserRouter>
+    );
+    expect(await screen.findByTestId('items-cards-list')).toBeInTheDocument();
+
+    fakeFetch = jest.spyOn(global, 'fetch').mockImplementation(async () => {
+      return new Promise(async (resolve) => {
+        const sleep = async (delay: number) => setTimeout(() => {}, delay);
+        await sleep(1000);
+        resolve({
+          ok: true,
+          status: 200,
+          statusText: 'OK',
+          json: () => Promise.resolve({ docs: [], pages: 0 }),
+        } as Response);
+      });
+    });
+
+    userEvent.type(screen.getByRole('textbox'), '{selectall}Search without result');
+    userEvent.click(screen.getAllByRole('button')[0]);
+    expect(await screen.findByAltText('Loading...')).toBeInTheDocument();
+    expect(await screen.findByTestId('items-cards-list')).toBeInTheDocument();
+    expect(screen.queryByAltText('Loading...')).not.toBeInTheDocument();
+    expect(screen.queryAllByTestId('item-card').length).toBe(0);
+    expect(screen.queryByText('Nothing has been found')).toBeInTheDocument();
+    expect(fakeFetch.mock.calls[1]).toEqual([
+      'https://the-one-api.dev/v2/character/?page=1&limit=10&name=/Search without result/i',
+      options,
+    ]);
+  });
+
+  it('recieves invalid data', async () => {
+    fakeFetch = jest.spyOn(global, 'fetch').mockImplementation(async () => {
```

Comment:
Too much boilerplate, use library or move it to some sort of mocks folder to reuse it